evm_loader/program/src/entrypoint.rs

```diff
@@ -214,10 +212,10 @@ fn process_instruction<'a>(
             let mut storage = StorageAccount::new(storage_info, accounts, from_addr, trx.nonce)?;
 
             if trx.to.is_some() {
-                do_partial_call(&mut storage, step_count, &account_storage, accounts, trx.call_data)?;
+                do_partial_call(&mut storage, step_count, &account_storage, accounts, trx.call_data, trx.gas_limit.as_u64())?;
```

Comment:
`as_u64()` can panic. Prefer `try_from/try_into` for conversions

Reply:
Excellent point. Reworked to u64::try_from here and everywhere as_u64 had been used.